src/gallery/slider.js

```diff
@@ -99,13 +100,26 @@ const clientWidth = window.innerWidth,
 			errorElement = container.querySelector( `.${prefixClassname}-item-loading-error` )
 
 		if ( refresh ) {
-			imageElement.src = `${imageElement.src}?timestamp=${Date.now()}`
+			const slider = parentContainer.querySelector( `.${prefixClassname}` ),
+				items = slider.querySelectorAll( `.${prefixClassname}-item` )
+
+			items.forEach( item => {
+				const image = item.querySelector( 'img' )
+				if ( image ) {
+					item.removeChild( image )
+				}
+			} )
```

Comment:
the event and refresh are for the individual image refresh purpose, why are we remove all the images within the slider?

Reply:
Right we were originally refreshing each individual image and that's why we were seeing the behavior that [Pita described](https://github.com/wikimedia/wikipedia-preview/pull/70#issuecomment-700316653). The idea here now is to accomplish a more comprehensive refresh by triggering a `renderNext` which consequently triggers three `showImageAndInfo` calls with different offsets (-1, 0, 1). For the `renderNext` --> `showImageAndInfo` cycle to complete successfully, we have to remove the `img` elements that are being replaced (supposedly failed images becuase of the connection issues). I'm opting for removing all of the `img` elements just to be on the safe side and ensure that there will be no `img` elements conflicts. If the network is back on, all images would be successfully loaded again. Does that make sense?

Reply:
alright, this can make sure it loads all the images so user doesn't have to refresh each image repeatedly, which is not bad.